R/utils-ci.R

```diff
@@ -25,7 +25,27 @@ determine_license_env <- function(license) {
   }
 }
 
-compose_start <- function(connect_license = Sys.getenv("RSC_LICENSE"), rsc_version, clean = TRUE) {
+version_to_docker_tag <- function(version) {
+  # Prior to 2022.09.0, the plain version number was the tag
+  # After, it's "<ubuntu-codename>-<version>"
+  # If you want a specific image tag, just pass it in and it will go through unchanged
+  try(
+    {
+      numver <- numeric_version(version)
+      if (numver > "2023.06") {
+        version <- paste0("jammy-", version)
+      } else if (numver >= "2022.09") {
+        # There's both jammy and bionic for these, but the bionic ones
+        # seem more reliable in CI
+        version <- paste0("bionic-", version)
+      }
+    },
+    silent = TRUE
+  )
+  version
+}
+
```

Comment:
At first I thought this might be over-complicated, but it's good, and we will soon have `noble-` to deal with too.

Reply:
This is really just sugar; if you know the image tag you can just specify it (now, that is, bionic was hard-coded before). This just prevents us from having to remember whether it's `bionic` or `jammy` or whatever else.